addons/web/static/src/core/utils/colors.js

```diff
@@ -198,6 +198,111 @@ export function convertCSSColorToRgba(cssColor) {
+    // contexts related to specifying color and its associated parameters.
+
+    const regexPredefinedRgbParams = `(${regexPredefinedRgb})\\s+(?<c1>${regexAlphaValue}|none)\\s+(?<c2>${regexAlphaValue}|none)\\s+(?<c3>${regexAlphaValue}|none)`;
+
+    // <xyz-space> is a non-terminal symbol, which is a placeholder for a value
+    // that can be one of the predefined options listed.
+    // - "xyz" typically refers to the CIE 1931 XYZ color space, which is a
+    // widely used color space in color science and digital imaging.
+    // - "xyz-d50" indicates the CIE 1931 XYZ color space with a reference
+    // illuminant of D50. Illuminant D50 is a standardized light source used
+    // in color management, and this version of the XYZ color space is adapted
+    // to it.
+    // - "xyz-d65" indicates the CIE 1931 XYZ color space with a reference
+    // illuminant of D65, another standardized light source often used for
+    // color reproduction and calibration.
+
+    const regexXyzSpace = "(?<xyz_space>xyz|xyz\\-d50|xyz\\-d65)";
+
+    // <xyz-params> represents a set of parameters associated with a predefined
+    // XYZ color space. These parameters consist of the XYZ color space name,
+    // followed by three values, where each value can be either a number,
+    // a percentage, or "none.".
+
+    const regexXyzParams = `(${regexXyzSpace})\\s+(?<cc1>${regexAlphaValue}|none)\\s+(?<cc2>${regexAlphaValue}|none)\\s+(?<cc3>${regexAlphaValue}|none)`;
+
+    // <colorspace-params> can be one of two options: <predefined-rgb-params> or
+    // <xyz-params>.
+
+    const regexColorspaceParams = `(?<colorspace_params>${regexPredefinedRgbParams}|${regexXyzParams})`;
+
+    // <slash> = /
+
+    const regexSlash = "/";
+
+    // "<color()>" allows to define a color specification in a formalized
+    // manner. It starts with the "color(" keyword, specifies color space
+    // parameters, and optionally includes an alpha value for transparency.
+
+    const color = `color\\(${regexColorspaceParams}(\\s+${regexSlash}\\s+(?<A>${regexAlphaValue}|none))?\\);?`;
+    const regex = new RegExp(color);
+
+    if (regex.test(cssColor)) {
+        // We can use regex.exec(cssColor).groups to actually perform the real
+        // calculations to convert color spaces to RGBa. But this trick is
+        // relatively robust. We'll see if we can define self-contained
+        // conversions if it doesn't fit to performance goals.
+
+        const canvasEl = document.createElement("canvas");
+        canvasEl.height = 1;
+        canvasEl.width = 1;
+        const ctx = canvasEl.getContext("2d");
+        ctx.fillStyle = cssColor;
+        ctx.fillRect(0, 0, 1, 1);
+        const data = ctx.getImageData(0, 0, 1, 1).data;
+        return {
+            red: data[0],
+            green: data[1],
+            blue: data[2],
+            opacity: parseInt(data[3] / 2.55), // Convert 0-255 to percentage
+        };
+    }
```

Comment:
Following the investigation, it has been observed that this function yields "false" when the variable does not correspond to a color. Consequently, reliance on the getImageData method or the fillStyle property is not viable as they exhibit leniency, returning #000000 for any non-understood string. Therefore, a recourse to Regex for validating the css color() function is essential to maintain code integrity. For instance, if a variable holds the value "Roboto," the function should return "false" to preserve the variable's value. What do you think?

Reply:
To handle CSS colors represented in their named form, I could create a comprehensive table that catalogs these colors while intentionally excluding "white" and "black." to avoid collision. This compilation would aid in their conversion to the RGBA format. However, the question remains: is this approach a definitive solution?

Reply:
> returning #000000 for any non-understood string

I think it does not go to black, it stays the way it is. So I guess a solution could be "set the color, if it changed, then it's a valid color".

Otherwise if not that, I think just checking that the value is in the form `color(XXX)` without validating `XXX` is enough. In case someone has a legitimate reason to send "color(random stuff)" and expect the function to return false, it won't work (it will return black as you said) but it is very unlikely. The easier the code the better IMO. And in that case, we can forget about the named CSS colors, it's not important.